src/widgets/chart.rs

```diff
@@ -362,7 +375,17 @@ impl<'a> Chart<'a> {
             .unwrap_or_default();
         if let Some(ref x_labels) = self.x_axis.labels {
             if !x_labels.is_empty() {
-                max_width = max(max_width, x_labels[0].content.width() as u16);
+                let first_label_width = x_labels[0].content.width() as u16;
```

Comment:
```rust
if let Some(first_label_width) = x_labels.get(0).map(|l| l.content.width() as u16) {
  //
}
```
maybe ? to avoid the empty test ?

Reply:
It might just be me, but I feel like too much is going on in this single line.
How about merging the two nested conditions ?

```rust
if let Some(first_x_label) = self.x_axis.labels.as_ref().and_then(|labels| labels.get(0)) {
    let first_label_width = first_x_label.content.width() as u16; 
    //
}
```